tests/integration/targets/aci_hsrp_interface_policy/tasks/main.yml

```diff
@@ -0,0 +1,154 @@
+      - nm_add_hsrp is changed
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.ctrl == "bia"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.delay == "2"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.reloadDelay == "2"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.name == "hsrp1"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.dn == "uni/tn-ansible_test/hsrpIfPol-hsrp1"
+
+  - name: Add the HSRP interface policy again to check idempotency
+    cisco.aci.aci_hsrp_interface_policy: *hsrp_present
+    register: nm_add_hsrp_again
+
+  - name: Verify add_hsrp again
+    assert:
+      that:
+      - nm_add_hsrp_again is not changed
+
+  - name: Update hsrp interface policy
+    cisco.aci.aci_hsrp_interface_policy: &hsrp_update
+      <<: *hsrp_present
+      delay: 3
+    register: nm_hsrp_update
+
+  - name: Verify update hsrp
+    assert:
+      that:
+      - nm_hsrp_update is changed
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.delay == "3"
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.reloadDelay == "2"
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.name == "hsrp1"
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.dn == "uni/tn-ansible_test/hsrpIfPol-hsrp1"
+
+  - name: Update hsrp interface policy to check for errors in delay
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *hsrp_update
+      delay: 10001
+    register: nm_hsrp_error1
+    ignore_errors: true
+
+  - name: Update hsrp interface policy to check for errors in reload delay
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *hsrp_update
+      reload_delay: 10001
+    register: nm_hsrp_error2
+    ignore_errors: true
+
+  - name: Verify errors
+    assert:
+      that:
+      - nm_hsrp_error1.msg == "Parameter 'delay' is only valid in range between 1 and 10000."
+      - nm_hsrp_error2.msg == "Parameter 'reload_delay' is only valid in range between 1 and 10000."
+
+  - name: Add another HSRP interface policy in check mode
+    cisco.aci.aci_hsrp_interface_policy: &hsrp2
+      <<: *tenant_present
+      hsrp: hsrp2
+      controls: ["bfd"]
+      state: present
+    register: add_hsrp2
+
+  - name: Query hsrp interface policy 1
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *hsrp_update
+      state: query
+    register: query_hsrp
+
+  - name: Query all hsrp interface policies
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *tenant_present
+      state: query
+    register: query_all
+
+  - name: Verify query hsrp
+    assert:
+      that:
+      - query_hsrp is not changed
+      - query_all is not changed
+      - query_hsrp.current.0.hsrpIfPol.attributes.delay == "3"
+      - query_hsrp.current.0.hsrpIfPol.attributes.reloadDelay == "2"
+      - query_hsrp.current.0.hsrpIfPol.attributes.name == "hsrp1"
+      - query_hsrp.current.0.hsrpIfPol.attributes.dn == "uni/tn-ansible_test/hsrpIfPol-hsrp1"
+      - query_all.current.0.fvTenant.children | length == 2
+      
+  - name: Remove hsrp interface policy 1
```

Comment:
Should we add missing check mode for removal?

Comment applies for other modules also.

Reply:
Ok